mantidproject/mantid
# Fix deselecting a first subcategory in Pref->Mantid->Options unticks main category

Description of work.
Certain categories under "View->Preferences->Mantid->Options" would take the value of their first subcategory depending on if they themselves were a category. 

When we get the list of categories and subcategories we now recurse backwards checking for any parent categories which were first seen from a subcategory and default them to a tick on as they do not have a visibility property associated.

**To test:**
Open "View->Preferences->Mantid->Options" all categories and subcategories should render correctly.
Categories such as "Algorithm" will preserve the tick option as it is standalone. The "Crystal" category is derived from its sub categories so will always appear ticked. 
Ensure the users selection is preserved across applying and re-opening the settings window.

Fixes #16871 .

Release notes have been updated

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
